drivers/platform/mbed/mbed_uart.cpp

```diff
@@ -0,0 +1,496 @@
+	{
+		return this->usb_iserial_descriptor;
+	}
+
+	void change_terminal_connection(bool connect_status);
+	bool data_received(uint32_t rx_size);
+	bool data_transmited(void);
+};
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Change the terminal connection status (for non-terminal) based USB interface
+ * @param connect_status- new connection status
+ * @note  This functions is used to change the terminal connection status of USB client
+ *        interface which is different than the 'console terminal'. The console terminals acknowledge
+ *        back to USB host when USB connection is opened on the console terminal and Mbed USBCDC
+ *        class automatically changes the '_terminal_connected' status accordingly. However, for
+ *        custom PC applications (non terminal), the terminal connection status needs to be changed
+ *        manually. The '_terminal_connected' is protected member of USBCDC parent class and thus can
+ *        be accessed through 'platform_usbcdc' derived class using below function.
+ */
+void platform_usbcdc::change_terminal_connection(bool connect_status)
+{
+	_terminal_connected = connect_status;
+}
+
+/**
+ * @brief	Check if new USB data is received/available
+ * @param	bytes[in] - Number of expected bytes to be received
+ * @return	true if expected number of bytes received, else false
+ */
+bool platform_usbcdc::data_received(uint32_t bytes)
+{
+	volatile static uint32_t *rx_size = &_rx_size;
+
+	if (*rx_size >= bytes)
+		return true;
+	else
+		return false;
+}
+
+/**
+ * @brief  Check if old USB data was transmitted
+ * @return true if transmit not in progress, else false
+ */
+bool platform_usbcdc::data_transmited(void)
+{
+	volatile static bool *tx_in_progress = &_tx_in_progress;
+
+	if (*tx_in_progress)
+		return false;
+	else
+		return true;
+}
+
+/**
+ * @brief Read data from UART device.
+ * @param desc[in, out] - Instance of UART.
+ * @param data[out] - Pointer to buffer containing data.
+ * @param bytes_number[in] - Number of bytes to read.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t uart_read(struct uart_desc *desc, uint8_t *data,
+		  uint32_t bytes_number)
+{
+	uint32_t cnt;
+	mbed::BufferedSerial *uart;	// pointer to BufferedSerial/UART instance
+	platform_usbcdc *usb_cdc_dev;	// Pointer to usb cdc device class instance
+	uint32_t size_rd;
+
+	if (!desc || !desc->extra || !data)
+		return -EINVAL;
+
+	if (!((struct mbed_uart_desc *)(desc->extra))->uart_port)
+		return -EINVAL;
+
+	if (((struct mbed_uart_desc *)desc->extra)->virtual_com_enable) {
+		usb_cdc_dev = (platform_usbcdc *)((struct mbed_uart_desc *)(
+				desc->extra))->uart_port;
+
+		while (!usb_cdc_dev->data_received(bytes_number)) {
+			/* Wait until new data is available */
+		}
```

Comment:
Not necessary right now, but it might be good to have a timeout.

Reply:
This is blocking read so not sure if we need timeout? Also it is difficult to estimate timeout value as it depends upon target MCU clock frequency and instruction set, so may be put it as MAX_INT32 if needed at all?